atomate/vasp/firetasks/parse_outputs.py

```diff
@@ -708,6 +708,134 @@ def run_task(self, fw_spec):
         logger.info("Thermal expansion coefficient calculation complete.")
 
 
+@explicit_serialize
+class PolarizationToDb(FiretaskBase):
+    """
+    Recovers the same branch polarization and spontaneous polarization
+    for a ferroelectric workflow.
+    """
+
+    optional_params = ["db_file"]
+
+    def run_task(self, fw_spec):
+        import re
+        from pymatgen.analysis.ferroelectricity.polarization import Polarization, get_total_ionic_dipole, \
+            EnergyTrend
+
+        wfid = list(filter(lambda x: 'wfid' in x, fw_spec['tags'])).pop()
+        db_file = env_chk(self.get("db_file"), fw_spec)
+        vaspdb = VaspCalcDb.from_db_file(db_file, admin=True)
+
+        # ferroelectric workflow groups calculations by generated wfid tag
+        polarization_tasks = vaspdb.collection.find({"tags": wfid, "task_label": {"$regex": ".*polarization"}})
+
+        tasks = []
+        outcars = []
+        structure_dicts = []
+        sort_weight = []
+        energies_per_atom = []
+        energies = []
+        zval_dicts = []
+
+        for p in polarization_tasks:
+            # Grad data from each polarization task
+            c = p['calcs_reversed'][0]
+            t = p['task_label']
+            o = c['output']['outcar']
+            s = c['input']['structure']
+            e_a = c['output']['energy_per_atom']
+            e = c['output']['energy']
+
+            energies_per_atom.append(e_a)
+            energies.append(e)
+            tasks.append(t)
+            outcars.append(o)
+            structure_dicts.append(s)
+            zval_dicts.append(o['zval_dict'])
+
+            # Add weight for sorting
+            # Want polarization calculations in order of nonpolar to polar for Polarization object
+            if 'nonpolar_polarization' in t:
+                sort_weight.append(0)
+            elif "polar_polarization" in t:
+                sort_weight.append(1000000)
+            elif "interpolation_" in t:
+                num = 0
+                part = re.findall(r'interpolation_[0-9]+_polarization', t)
+                if part != []:
+                    part2 = re.findall(r'[0-9]+', part.pop())
+                    if part2 != []:
+                        num = part2.pop()
+                sort_weight.append(1000 - int(num))
+
+        # Sort polarization tasks
+        # nonpolar -> interpolation_n -> interpolation_n-1 -> ...  -> interpolation_1 -> polar
+        data = zip(tasks, structure_dicts, outcars, energies_per_atom, energies, sort_weight)
+        data.sort(key=lambda x: x[-1])
+
+        tasks, structure_dicts, outcars, energies_per_atom, energies, sort_weight = zip(*data)
```

Comment:
add a comment to help people follow - say that you are re-getting these variables in sorted order

Reply:
Done.